src/value.rs

```diff
@@ -204,22 +233,33 @@ impl ValueCommitment {
         // TODO: impl From<pallas::Point> for redpallas::VerificationKey.
         self.0.to_bytes().try_into().unwrap()
     }
+
+    /// Deserialize a value commitment from its byte representation
+    pub fn from_bytes(bytes: &[u8; 32]) -> CtOption<ValueCommitment> {
+        pallas::Point::from_bytes(bytes).map(ValueCommitment)
+    }
+
+    /// Serialize this value commitment to its canonical byte representation.
+    pub fn to_bytes(&self) -> [u8; 32] {
+        self.0.to_bytes()
+    }
 }
 
-#[cfg(test)]
-mod tests {
+/// Generators for property testing.
+#[cfg(any(test, feature = "test-dependencies"))]
+pub mod testing {
     use pasta_curves::{arithmetic::FieldExt, pallas};
     use proptest::prelude::*;
 
-    use super::{OverflowError, ValueCommitTrapdoor, ValueCommitment, ValueSum};
-    use crate::primitives::redpallas;
+    use super::{NoteValue, ValueCommitTrapdoor, ValueSum};
 
     /// Zcash's maximum money amount. Used as a bound in proptests so we don't artifically
     /// overflow `ValueSum`'s size.
-    const MAX_MONEY: i64 = 21_000_000 * 1_0000_0000;
+    pub const MAX_MONEY: i64 = 21_000_000 * 1_0000_0000;
```

Comment:
I'd prefer this was not part of the public API (even behind the `test-dependencies` feature flag).

Reply:
The bsk_consistent_with_bvk proptest looks wrong to me. Currently it has this:
```
        fn bsk_consistent_with_bvk(
            values in prop::collection::vec((arb_value_sum(MAX_MONEY), arb_trapdoor()), 1..10),
        ) {
            let value_balance = values
                .iter()
                .map(|(value, _)| value)
                .sum::<Result<ValueSum, OverflowError>>()
                .expect("we generate values that won't overflow");
```

This doesn't test the full range of ValueSum; also, it seems like the expectation that the overflow can't occur in the summation there would no longer be true if we were testing the full range of ValueSum (in the case, for example, that all max values were generated.